roles/btrfsmaintenance/tasks/main.yml

```diff
@@ -0,0 +1,79 @@
+#########################################################################
+# Title:         Community: btrfsmaintenance                            #
+# Author(s):     owine                                                  #
+# URL:           https://github.com/Cloudbox/Community                  #
+# --                                                                    #
+#         Part of the Cloudbox project: https://cloudbox.works          #
+#########################################################################
+#                   GNU General Public License v3.0                     #
+#########################################################################
+---
+- name: "Determine '/' filesystem type"
+  set_fact:
+    root_fstype: "{{ item.fstype }}"
+  when: (item.mount == '/')
+  with_items:
+    - "{{ ansible_mounts }}"
+
+- name: "Determine '/opt' filesystem type"
+  set_fact:
+    opt_fstype: "{{ item.fstype }}"
+  when: (item.mount == '/opt')
+  with_items:
+    - "{{ ansible_mounts }}"
+
+- name: "Setup btrfsmaintenance scripts"
+  block:
+
+  - name: "Clone btrfsmaintenance repo"
+    git:
+      repo: https://github.com/kdave/btrfsmaintenance.git
+      dest: '/opt/btrfsmaintenance'
+      clone: yes
+      version: HEAD
+      force: yes
+    become: yes
+    become_user: "{{ user.name }}"
+
+  - name: "Set btrfsmaintenance directory permissions"
+    file: "path='/opt/btrfsmaintenance' state=directory mode=0775 owner={{ user.name }} group={{ user.name }} recurse=yes"
+
+  - name: "Change mountpoints to 'auto'"
+    lineinfile:
+      path: "/opt/btrfsmaintenance/sysconfig.btrfsmaintenance"
+      regexp: '^BTRFS_TRIM_MOUNTPOINTS\s?='
+      line: 'BTRFS_TRIM_MOUNTPOINTS="auto"'
+      state: present
+
+  - name: "Execute dist-install.sh"
+    become: yes
+    become_user: root
+    shell: dist-install.sh
+    args:
+      chdir: /opt/btrfsmaintenance/
+
+  - name: "Copy .timer files to systemd"
+    copy:
+     src: /opt/btrfsmaintenance/*.timer
+     dest: /etc/systemd/system
+    delegate_to: localhost
+    
+  - name: "Copy .service files to systemd"
+    copy:
+     src: /opt/btrfsmaintenance/*.service
+     dest: /etc/systemd/system
+    delegate_to: localhost
+
+  - name: "Execute btrfsmaintenance-refresh-cron.sh"
+    become: yes
+    become_user: root
+    shell: btrfsmaintenance-refresh-cron.sh timer
```

Comment:
I assume this will need the full path somewhere too, otherwise it will fail similar to the above.